cmd/net.go

```diff
@@ -66,18 +67,37 @@ func mustGetLocalIP4() (ipList set.StringSet) {
 // getHostIP4 returns IPv4 address of given host.
 func getHostIP4(host string) (ipList set.StringSet, err error) {
 	ipList = set.NewStringSet()
-	ips, err := net.LookupIP(host)
-	if err != nil {
-		return ipList, err
-	}
-
-	for _, ip := range ips {
-		if ip.To4() != nil {
-			ipList.Add(ip.String())
+	//variable to list resolved IP addresses
+	var ips []net.IP
+	// channel to indicate completion of host resolution
+	doneCh := make(chan struct{})
+	// Indicate to our retry routine to exit cleanly, upon this function return.
+	defer close(doneCh)
+	// Wait on the jitter retry loop
+	retryTimeCh := newRetryTimerSimple(doneCh)
+	for {
+		select {
+		case <-retryTimeCh:
+			// try to resolve remote host
+			ips, err = net.LookupIP(host)
+			if err != nil {
+				// log error and continue. We dont fail on host resolution, just log
+				// the message to console about the host not being resolveable. It is
+				// expected that user provides resolveable hosts
+				errorIf(err, "Unable to resolve host %s (elapsed %s)", host, getElapsedTime(time.Now().Round(time.Second)))
+			} else {
+				// add to list if successfully resolved
+				for _, ip := range ips {
+					if ip.To4() != nil {
+						ipList.Add(ip.String())
+					}
+				}
+				return ipList, err
+			}
+		case <-globalServiceDoneCh:
```

Comment:
Why do you want to handle this channel here? For this function its not required.

below code is enough
```go
doneCh := make(chan struct{})
defer close(doneCh)
var ips []net.IP
for _ := range newRetryTimerSimple(doneCh) {
    ips, err := net.LookupIP(host)
    errorIf(err, "Unable to resolve host %s (elapsed %s)", host, getElapsedTime(time.Now().Round(time.Second)))
    if err == nil {
        break
    }
}
```